prog_test.go

```diff
@@ -547,21 +547,47 @@ func TestProgramSpecTag(t *testing.T) {
 }
 
 func TestProgramTypeLSM(t *testing.T) {
-	prog, err := NewProgram(&ProgramSpec{
-		AttachTo:   "task_getpgid",
-		AttachType: AttachLSMMac,
-		Instructions: asm.Instructions{
-			asm.LoadImm(asm.R0, 0, asm.DWord),
-			asm.Return(),
+	lsmTests := []struct {
+		attachFn    string
+		flags       uint32
+		expectedErr bool
+	}{
+		{
+			attachFn: "task_getpgid",
+		},
+		{
+			attachFn:    "task_setnice",
+			flags:       unix.BPF_F_SLEEPABLE,
```

Comment:
These will have to be gated with `testutils.SkipOnOldKernel` since I'm guessing that BPF_F_SLEEPABLE was introduced at a different time than LSM programs. Maybe easier to have `sleepable bool` in the test definition?

Reply:
added a kernel version check against 5.10